### ISSUE TITLE
Uncaught SyntaxError: Unexpected token }

### ISSUE THREAD

**Issue by tomitrescak** [fuse-box/fuse-box#325]
Hi, very often after application does stateful reload I get error `Uncaught SyntaxError: Unexpected token }`

Afterwards, the only way to get rid of this error is to delete the cache and rebuild.

This is in the source:

```js
FuseBox.import("fusebox-hot-reload").connect(5680, "")

FuseBox.import("default/client/modules/_stories/stories.jsx");
FuseBox.main("default/client/modules/_stories/stories.jsx");
}) 
... // FB stuff
//# sourceMappingURL=/libs/luis/luis.js.map

FuseBox.import("default/client/modules/_stories/stories.jsx");
FuseBox.main("default/client/modules/_stories/stories.jsx");
}) // this is where error is
.. // FB Stuff
//# sourceMappingURL=/libs/luis/luis.js.map
```

It seems that it is imported somehow twice.

**Comment by nchanged** [comment]
Are you using the latest version?

**Comment by nchanged** [comment]
And another question, how many bundles are doing at the same time? Do they share the same code?

**Comment by tomitrescak** [comment]
@nchanged I am creating 2 bundles (vendor.js and app.js), they share only `+pretty-format +fuse-box`.

**Comment by nchanged** [comment]
@tomitrescak can we have a repository that could reproduce this bug? I know it might happen, it's related to caching. It's a very important thing to resolve. 

Thanks!

**Comment by tomitrescak** [comment]
@nchanged unfortunately the one I use is private, but am starting a new OSS project with FuseBox so as soon as the error appears there I'd let you know.

**Comment by tomitrescak** [comment]
Guys, It happened in this repo as well: https://github.com/tomitrescak/luis-app
After I worked with it a bit (added / removed files, added removed CSS references) I got the very same error.

**Comment by nchanged** [comment]
Hm.. so a way to reproduce it would be?

* Add file
* Remove
* Add again

?

**Comment by tomitrescak** [comment]
Try removing reference in ts file .. adding reference ...
then you can try adding and removing file ...

IO am getiing it now quite often.

**Comment by nchanged** [comment]
Ok just CSS - is fine.
![css-test](https://cloud.githubusercontent.com/assets/1629088/23543585/1ae127b0-fff3-11e6-958d-9b6c1fac8e8e.gif)

**Comment by nchanged** [comment]
Can you install `fsbx@1.3.123-preview.10` please? 
```
require("fsbx")
```

That's the latest in the pipeline.

**Comment by tomitrescak** [comment]
Try full page reload afterwards. that's when it explodes.

**Comment by nchanged** [comment]
Nope, it does not =)
Can you try the latest preview?

**Comment by tomitrescak** [comment]
Did you try it with that repo I provided? On a simple app it obviously works well. Even with a preview it is still bringing up the error ;(

**Comment by nchanged** [comment]
Can't seem to launch it .. I guess you are missing index.html in your build.
It says `Cannot Get /`

**Comment by tomitrescak** [comment]
Can you try again, users reported that everything works fine. In my project it's geting increasingly bad. Now I get the error every 3rd - 4th HMR ;( And I'm not doing anything fancy.

**Comment by tomitrescak** [comment]
Guys .. I can confirm that ERROR no longer appears in 1.3.123 !!

BUT

I had to downgrade back to 1.2.122 as 1.3.123 breaks the bundle after HMR.
When I do full page relaod after HMR, I get following error:

```
process is not defined
```

It seems that rebuilt bundle does not execute something that needs to be executed. The only way to make the bundle work again is to delete the .fusebox directory and rebuild.

**Comment by nchanged** [comment]
@tomitrescak gotha. Something i've been "fixing" 2 times already... Will check it again ;-) Thanks

**Comment by mattoni** [comment]
I'd like to add that I'm also experiencing this issue occasionally with the SSR demo I'm working on.

https://github.com/mattoni/fusemob-ssr

(Also getting this: `DevTools failed to parse SourceMap: http://localhost:3000/index.js.map`)

To reproduce, go into the components/Header and change the BG color, then do a hard refresh. I think there's a mismatch or something that's coming from the server. If I kill it and recompile after receiving the syntax error, I get this:
```
Uncaught TypeError: Cannot read property 'render' of undefined
```

as it looks like now ReactDOM isn't bundled anymore. Only way to fix is to delete .fusebox folder and recompile. 

Hopefully this gives you another thing to test with and help. Let me know if I can provide any more information.

**Comment by nchanged** [comment]
@mattoni are you a windows user?

What I get at max is:

<img width="1290" alt="screenshot 2017-03-13 19 49 24" src="https://cloud.githubusercontent.com/assets/1629088/23870282/3e07c74a-0826-11e7-8358-da2a5b2b433d.png">

**Comment by mattoni** [comment]
I am on OSX, and that happens because I'm not restarting the server when I make a change to the bundle, so it's serving the old React (trying to figure out a workaround to that). If I restart the server on every change, everything falls apart (to see what I mean, change the watch tree syntax in fuse.ts to watch ./src instead of ./src/server).

Let me see if I can get some repeatable steps for you.

**Comment by mattoni** [comment]
So, from a fresh cold start (deleting .fusebox folder), things work. Then I change a color or some style in a component. It does the HMR just fine, but the console shows this:

![image](https://cloud.githubusercontent.com/assets/5110855/23870535/eee1222a-07e3-11e7-9c1d-c677cef56eca.png)

Then, I do a hard refresh and get the same error you do (because the server is not sending the right information). 

I change the colors once more, and do a hard refresh. Now the syntax error shows up.

![image](https://cloud.githubusercontent.com/assets/5110855/23870587/1dcfecd8-07e4-11e7-8657-3c8d36c6da22.png)

From here, even killing and rebuilding wont work, since react dom doesnt get included. 

![image](https://cloud.githubusercontent.com/assets/5110855/23870619/36d2d506-07e4-11e7-8d3e-c7a0aebf8237.png)

Only way to continue is to delete .fusebox folder and rebuild.

**Comment by devmondo** [comment]
this might be related to cache bug we are investigating, we get same thing on windows, it seems a race condition where file is not written correctly.

**Comment by nchanged** [comment]
<img width="784" alt="screenshot 2017-03-13 20 15 49" src="https://cloud.githubusercontent.com/assets/1629088/23871252/e437fa1a-0829-11e7-9892-30849ae62a71.png">
Quite interesting, it says that's it's not found, however i see it in the `bundle.js` But not when inspecting the page and viewing the source...

**Comment by nchanged** [comment]
I see the problem.. It your drop the cache, you will see
```
FuseBox.pkg("react-dom", {}, function(___scope___){
___scope___.file("index.js", function(exports, require, module, __filename, __dirname){

'use strict';

module.exports = require('./lib/ReactDOM');

});
```
Which is react for browser, but next time it's the screenshot above!!

**Comment by mattoni** [comment]
Hmm, does that mean it's a problem having two FuseBox builds using the same cache?

**Comment by nchanged** [comment]
That's the thing .. it should not happen, but they overlap for some reason.

**Comment by mattoni** [comment]
Let me know if you need anything else to test it effectively. I unfortunately won't have time to dive down into this for a couple days since I'm not familiar with the fusebox code base yet, but I'd be glad to help out this weekend. 

I'm looking to build quite a bit of my company's front end with this repo I'm working on, so let me know as soon as there's a fix available :) I don't want to switch back to webpack.

**Comment by nchanged** [comment]
I hope I will fix it tomorrow as it's a very dangerous bug. Will keep you posted.

**Comment by nchanged** [comment]
I think this bug is related to the broken Sass build... will get on it very soon.

**Comment by nchanged** [comment]
Please feel free to check the latest bleeding edge version `fuse-box@2.0.0-beta.8`
and documentation for it http://fuse-box.org:3333 ;-)

**Comment by devmondo** [comment]
this is supposed to be fixed in version 2.0

we will close this for now, but feel free to open it otherwise :)
